CommonMark/Formatters/HtmlFormatter.cs

```diff
@@ -495,6 +497,40 @@ protected virtual void WriteInline(Inline inline, bool isOpening, bool isClosing
                     }
                     break;
 
+                case InlineTag.Placeholder:
+                    ignoreChildNodes = false;
+
+                    if (isOpening)
+                    {
+                        string placeholderSubstitute = (_placeholderResolver != null) ? _placeholderResolver(inline.TargetUrl) : null;
```

Comment:
If I remember correctly (didn't check right now), the other callbacks that the user provide are wrapped in try/catch and throws a wrapped exception - since otherwise people tend to assume that the exception is caused by CommonMark.NET instead of the user code.